src/test/java/uk/nhs/digital/nhsconnect/lab/results/model/edifact/LaboratoryInvestigationResultTest.java

```diff
@@ -0,0 +1,105 @@
+package uk.nhs.digital.nhsconnect.lab.results.model.edifact;
+
+import org.junit.jupiter.api.Test;
+import uk.nhs.digital.nhsconnect.lab.results.model.edifact.message.EdifactValidationException;
+
+import java.math.BigDecimal;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.SoftAssertions.assertSoftly;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+class LaboratoryInvestigationResultTest {
+
+    private final LaboratoryInvestigationResult laboratoryInvestigationResult1 = new LaboratoryInvestigationResult(
+            new BigDecimal("11.9"), MeasurementValueComparator.LESS_THAN, "ng/mL", DeviatingResultIndicator.ABOVE_HIGH_REFERENCE_LIMIT
+    );
+    private final LaboratoryInvestigationResult laboratoryInvestigationResult2 = new LaboratoryInvestigationResult(
+            new BigDecimal("11.9"), null, "ng/mL", DeviatingResultIndicator.ABOVE_HIGH_REFERENCE_LIMIT
+    );
+    private final LaboratoryInvestigationResult laboratoryInvestigationResult3 = new LaboratoryInvestigationResult(
+            new BigDecimal("11.9"), null, "ng/mL", null
+    );
+
+    @Test
+    void when_edifactStringDoesNotStartWithLaboratoryInvestigationResultKey_expect_illegalArgumentExceptionIsThrown() {
+        assertThrows(IllegalArgumentException.class, () -> LaboratoryInvestigationResult.fromString("wrong value"));
+    }
+
+    @Test
+    void when_edifactString1IsPassed_expect_returnALaboratoryInvestigationResultObject() {
+        assertThat(laboratoryInvestigationResult1)
```

Comment:
An approach to unit tests that I like is to write them so that the reader doesn't have to look anywhere else except the method describing the test. In these cases, there's no obvious reason why `laboratoryInvestigationResult1` should end up having anything to do with `"RSL+NV+11.9:7++:::ng/mL+HI"` without going and seeing how the constant object was constructed.

Personally I'd prefer these to be inlined, or constructed within the method itself.

Reply:
Yeah makes sense. I can make that change for `laboratoryInvestigationResult2` and `laboratoryInvestigationResult3`, but since `laboratoryInvestigationResult1` is shared between tests I'll keep it at the top of the file

Reply:
I'd argue the fact they're the same is incidental. You could use other data and get the same benefits of having everything in the same method. In DRY terms, they are "distinct pieces of knowledge", so even if they have the same representation, it's correct for them to have different representation.